resources/providers/azurelib/inventory/storage_provider.go

```diff
@@ -77,9 +86,52 @@ func NewStorageAccountProvider(log *logp.Logger, diagnosticSettingsClient *armmo
 	}
 }
 
-func (p *storageAccountProvider) ListStorageAccountBlobServices(ctx context.Context, storageAccounts []AzureAsset) ([]AzureAsset, error) {
+func (p *storageAccountProvider) ListStorageAccounts(ctx context.Context, storageAccountsSubscriptionsIds []string) ([]AzureAsset, error) {
 	var assets []AzureAsset
+	for _, saID := range storageAccountsSubscriptionsIds {
+		res, err := p.client.AssetAccountStorage(ctx, saID, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error while fetching storage accounts: %w", err)
+		}
+		storageAccountsAssets, err := transformStorageAccounts(res, saID)
+		if err != nil {
+			return nil, fmt.Errorf("error while transforming storage accounts: %w", err)
+		}
+		assets = append(assets, storageAccountsAssets...)
+	}
+	return assets, nil
+}
 
+func transformStorageAccounts(accountPages []armstorage.AccountsClientListResponse, storageAccountId string) (_ []AzureAsset, errs error) {
```

Comment:
I see you used name returns as a mean to declare errs with value zero (`nil`).

This saves one line, but not sure if it helps on readability. I would favor by manually init the errs variable manually tbh. I don't remember seeing usage of name returns for the purpose you used here.